# MSc-Project: Using single-cell RNA sequencing to identify immune-cell population
### Abstract 
Single-cell RNA sequencing (scRNA-seq) technology is incredibly powerful tool to study the intricacy and dynamic nature of biological process, as it can provide transcriptomes at a single-cell level. It can also provide unparalleled resolution for cell identification and discovery for clinical application, such as cancer diagnostic.However, analyzing and interpreting scRNA-seq result can be challenging, as the expression data becomes bimodal with high background noise. Conventional cell classification methods used in Fluorescence-activated cell sorting (FACS) and bulk RNA-sequencing do not adequately address the problems. Here, a marker-based classification is developed to identify known cell type, defined by the cell-surface protein markers, for fresh peripheral blood mononuclear cells (PBMCs). This classification approach is then compared against the correlation-based approaches, used in the secondary analysis of 10X Genomics Chromuim scRNA-seq platform, in both 68k fresh PBMC sample and simulation. Even though the correlation-based approaches outperforms in both data sets, the marker-based approach can still provide an adequate classification.However, both approaches failed to differential among T-cell subgroups which show  cellular heterogeneity remains a significant challenges in scRNA-seq analysis.Finally, to address the shortcomings of scRNA-seq data, two additional scRNA-seq specific tools are used , but no improvement is observed.

## Analysis and Figures used in the thesis 
All codes used in analysis and figures is under final_analysis folder

## Data used in thesis 
All data are obtained from https://github.com/sfpacman/single-cell-3prime-paper/tree/master/pbmc68k_analysis#single-cell-rna-seq-secondary-analysis-of-68k-pbmcs


R-markdown HTML page can be found at( Update in prgress):
http://student.cryst.bbk.ac.uk/~yp001/project_index.html
