<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
    "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<title> MSc Master Project</title>
<!--<link rel="shortcut icon" href="/home/yup/bioinfo/biocomp2/icon.png" type="image/x-icon" /> -->
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
<!-- jQuery library -->
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
<!-- Latest compiled JavaScript -->
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1"> 
<style type="text/css">

  body { color: #65A3FF; background: white; overflow-x: hidden;}
.non-floater { overflow:hidden; }
iframe { width:100%; padding:0; border:0; height:100%; position: absolute;}
</style>
</head>
<body>
<div class="Plan">
<h2><center>Thesis Working Plan</center></h2>
<h3>Goal:</h3>
<ul>
<li> Adpot Danaher's RNA gene signature in scRNA-seq </li>
<li> Compare gene signatures from Danaher's and 10x 's apporach </li>
<li> determine whether using marker-based apporach is useful for scRNA-seq data</li>
</ul>
<h3>Steps:</h3>
<ul>
<li> Use scRNAseq-specific <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7">Normalizaton</a>+<a href="https://academic.oup.com/bioinformatics/article/31/12/1974/214505/Identification-of-cell-types-from-single-cell">Clustering method </a>to optimize the gene expression matrix for accurate comparsion </li>
<li> Adpot and optimize Danaher's method and apply to scRNAseq data obtained from 10x public dataset </li>
<li> Apply and compare gene signatures with 10x approach  </li>
<li> Examine which gene(s) "drive" the signature </li>
<li> Draw Conclusion whther Danaher's findings can be applied to scRNAseq data  </li>
</ul>
</div>
</body>